File: Competitions/HEMAYCL-16/LCM.py
def gcd (a,b):
  return a if b==0 else gcd(b,a%b)

def lcm (a,b):
  return a * (b / gcd(a,b))

def magic1(N):
  result = 1
  for i in range(2,N+1):
      result = lcm(result,i)
  return result

def magic2(M,N):
  result = 1
  for i in range(M,N+1):
      result = lcm(result,i)
  return result
for N in xrange(600,900):
    #N = int(raw_input())
    LCM = magic1(N)

    for i in xrange(N//2, N+1):
        check = magic2(i,N)
        #print i,"-",N," LCM: ",check;
        if check != LCM:
            break
    print N,": ",i-1," ",N//2

File: Competitions/HEMAYCL-16/LCM.cpp
#include <bits/stdc++.h>
using namespace std;

long long gcd(int a, int b)
{
    return b == 0 ? a : gcd(b, a % b);
}

long long lcm(long long a, long long b)
{
  return a * ( b / gcd (a,b));
}

long long do_magic(long long M, long long N)
{
  long long result = 1;
  for(int i = M;i <= N; i++)
  {
    result = lcm(result,i);
  }
  cout<<"From:"<<M<<"-"<<N<<endl;
  cout<<"LCM: "<<result<<endl;
}

long long do_magic(long long N)
{
  long long result = 1;
  for(int i = 2;i <= N; i++)
  {
    result = lcm(result,i);
  }
  cout<<"From 1 to "<<N<<endl;
  cout<<"LCM: "<<result<<endl;
}

int main()
{
  long long N,M;
  cin>>N;
  do_magic(N);
  cin>>M>>N;
  do_magic(M,N);
  return 0;
}
